staging/src/k8s.io/client-go/util/cert/pem.go

```diff
@@ -106,6 +107,64 @@ func ParsePrivateKeyPEM(keyData []byte) (interface{}, error) {
 	return nil, fmt.Errorf("data does not contain a valid RSA or ECDSA private key")
 }
 
+// ParsePublicKeysPEM is a helper function for reading an array of rsa.PublicKey or ecdsa.PublicKey from a PEM-encoded byte array.
+// Reads public keys from both public and private key files.
+func ParsePublicKeysPEM(keyData []byte) ([]interface{}, error) {
+	ok := false
+	var keys []interface{}
+	for len(keyData) > 0 {
+		var block *pem.Block
+		block, keyData = pem.Decode(keyData)
+		if block == nil {
+			break
+		}
+		// Only use PEM "PUBLIC KEY" or "PRIVATE KEY" blocks without extra headers
+		if block.Type != PublicKeyBlockType || block.Type != PrivateKeyBlockType || len(block.Headers) != 0 {
+			continue
+		}
+
+		switch block.Type {
+		case ECPrivateKeyBlockType:
+			privateKey, err := x509.ParseECPrivateKey(block.Bytes)
+			if err != nil {
+				return keys, err
+			}
+			keys = append(keys, &privateKey.PublicKey)
+		case RSAPrivateKeyBlockType:
+			privateKey, err := x509.ParsePKCS1PrivateKey(block.Bytes)
+			if err != nil {
+				return keys, err
+			}
+			keys = append(keys, &privateKey.PublicKey)
+		case PrivateKeyBlockType:
+			privateKey, err := x509.ParsePKCS8PrivateKey(block.Bytes)
+			if err != nil {
+				return keys, err
+			}
+			if parsedRSAKey, parseSuccess := privateKey.(*rsa.PrivateKey); parseSuccess {
+				keys = append(keys, &parsedRSAKey.PublicKey)
+			} else if parsedECKey, parseSuccess := privateKey.(*ecdsa.PrivateKey); parseSuccess {
+				keys = append(keys, &parsedECKey.PublicKey)
+			} else {
+				return keys, err
+			}
```

Comment:
So I did a bunch of looking into this, and while `RSA PUBLIC KEY` and `EC PUBLIC KEY` are valid PEM block types, it doesn't look like golang's https://golang.org/pkg/crypto/x509/#ParsePKIXPublicKey is set up to handle them. I generated an RSA PUBLIC KEY in that older format, and ParsePKIXPublicKey rejected it flat out. As ParsePKIXPublicKey was what jwt is using, we aren't losing functionality I don't think.